<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1128</width>
    <height>747</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Panorama Image Stitcher</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>70</y>
      <width>921</width>
      <height>611</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="img1Tab">
     <attribute name="title">
      <string>Image 1</string>
     </attribute>
     <widget class="QLabel" name="img1Display">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>12</y>
        <width>890</width>
        <height>565</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="img2Tab">
     <attribute name="title">
      <string>Image 2</string>
     </attribute>
     <widget class="QLabel" name="img2Display">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>12</y>
        <width>890</width>
        <height>565</height>
       </rect>
      </property>
      <property name="text">
       <string></string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="imgSTab">
     <attribute name="title">
      <string>Stitched Image</string>
     </attribute>
     <widget class="QLabel" name="imgSDisplay">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>12</y>
        <width>890</width>
        <height>565</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="openButton">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>20</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Open Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>20</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="QPushButton" name="harrisButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>100</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Harris Corners</string>
    </property>
   </widget>
   <widget class="QPushButton" name="RANSACButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>RANSAC</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stitchButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Stitch</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="harrisSpinBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="harrisThresSpinBox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>170</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>999.990000000000009</double>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Sigma</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>150</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Threshold</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="matchButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Find Matches</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="RANSACThresSpinBox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>410</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Iterations</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>390</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Threshold</string>
    </property>
   </widget>
  <widget class="QLabel" name="instructions_line0">
    <property name="geometry">
      <rect>
        <x>450</x>
        <y>10</y>
        <width>600</width>
        <height>15</height>
      </rect>
    </property>
    <property name="text">
      <string>On the Image 1 and Image 2 tabs, click Open Image to select images to stitch.</string>
    </property>
  </widget>
    <widget class="QLabel" name="instructions_line1">
      <property name="geometry">
        <rect>
          <x>450</x>
          <y>25</y>
          <width>600</width>
          <height>15</height>
        </rect>
      </property>
      <property name="text">
        <string>Click Harris Corners to detect corners in each image. Click Find Matches to guess corresponding corners.</string>
      </property>
    </widget>
    <widget class="QLabel" name="instructions_line2">
      <property name="geometry">
        <rect>
          <x>450</x>
          <y>40</y>
          <width>650</width>
          <height>15</height>
        </rect>
      </property>
      <property name="text">
        <string>Use RANSAC to eliminate outliers and compute a homography. Change settings/click RANSAC again if the homography looks wrong.</string>
      </property>
    </widget>
    <widget class="QLabel" name="instructions_line3">
      <property name="geometry">
        <rect>
          <x>450</x>
          <y>55</y>
          <width>600</width>
          <height>15</height>
        </rect>
      </property>
      <property name="text">
        <string>Click Stitch to stitch the images together! Click Save Image to save your stitched result.</string>
      </property>
    </widget>
    <widget class="QSpinBox" name="iterationsBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>9999</number>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1128</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
